### PR TITLE
[4852][IMP][REF] stock_quant_list_view, auction_seller

### PR DESCRIPTION
**Pull request by kanda999**
[4852](https://www.quartile.co/web#id=4852&cids=3&menu_id=505&action=1457&model=project.task&view_type=form)

- Refactoring the auction_seller module and migrating some functions to the product_seller module

- stock_quant_list_view module to display seller_id and allow grouping by yahoo_product_state_id

- yahoo_product_sate_id set to store=true in product_yahoo_auction_sst module

### CODE REVIEW

File: product_seller/models/stock_quant.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2024 Quartile
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    seller_id = fields.Many2one(related="product_id.seller_id")
```
Comment (yostashiro) [inline product_seller/models/stock_quant.py]: No `store=True`?

Comment (kanda999) [reply product_seller/models/stock_quant.py]: There is no reason to set store=false, so I added store=true.

File: auction_seller/__manifest__.py [hunk 2]
```diff
@@ -7,14 +7,13 @@
     "author": "Quartile Limited",
     "website": "https://www.quartile.co",
     "license": "LGPL-3",
-    "depends": ["product_auction", "event"],
+    "depends": ["product_auction", "product_seller", "event"],
```
Comment (yostashiro) [inline auction_seller/__manifest__.py]:
```suggestion
    "depends": ["event", "product_auction", "product_seller"],
```
Preferred order.

File: product_seller/models/stock_quant.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2024 Quartile
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    seller_id = fields.Many2one(related="product_id.seller_id", store=True)
```
Comment (yostashiro) [inline product_seller/models/stock_quant.py:12]: Shall we add a comment for future reference.
```suggestion
    # We assume that there is just a single (internal) quant per product; the performance risk of `store=True` is minimal.
    seller_id = fields.Many2one(related="product_id.seller_id", store=True)
```

File: product_yahoo_auction_sst/models/stock_quant.py [hunk 4]
```diff
@@ -10,4 +10,5 @@ class StockQuant(models.Model):
     yahoo_product_state_id = fields.Many2one(
         related="product_id.yahoo_product_state_id",
         string="Yahoo Product State",
+        store=True,
     )
```
Comment (yostashiro) [inline product_yahoo_auction_sst/models/stock_quant.py:16]:
```suggestion
    # We assume that there is just a single (internal) quant per product; the performance risk of `store=True` is minimal.
    yahoo_product_state_id = fields.Many2one(
        related="product_id.yahoo_product_state_id",
        string="Yahoo Product State",
        store=True,
    )
```

File: stock_quant_list_view/views/stock_quant_views.xml [hunk 5]
```diff
@@ -40,6 +41,23 @@
             </tree>
         </field>
     </record>
+    <record id="quant_search_view" model="ir.ui.view">
+        <field name="name">stock.quant.search</field>
+        <field name="model">stock.quant</field>
+        <field name="inherit_id" ref="stock.quant_search_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='product_id']" position="after">
+                <field name="seller_id" />
```
Comment (yostashiro) [inline stock_quant_list_view/views/stock_quant_views.xml:50]: Would be good to have this in 'group by' as well.

File: stock_quant_list_view/views/stock_quant_views.xml [hunk 6]
```diff
@@ -40,6 +41,23 @@
...
+            </xpath>
+            <xpath expr="//filter[@name='productgroup']" position="after">
+                <filter
+                    name="yahoo_product_state_id"
+                    string="Yahoo Product State"
+                    context="{'group_by':'yahoo_product_state_id'}"
+                />
```
Comment (yostashiro) [inline stock_quant_list_view/views/stock_quant_views.xml:63]: Would be good to have this in search as well.

File: product_yahoo_auction_sst/models/yahoo_product_state.py [hunk 7]
```diff
@@ -7,6 +7,8 @@
 class YahooProductState(models.Model):
     _name = "yahoo.product.state"
 
+    # We assume that there is just a single (internal) quant per product;
+    # the performance risk of `store=True` is minimal.
```
Comment (yostashiro) [inline product_yahoo_auction_sst/models/yahoo_product_state.py]: I think you put the comment in the wrong place.
```suggestion
```

**Review by yostashiro — APPROVED** [review]

**Pull request merged**